pkgs/misc/cups/libppd.nix

```
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, autoreconfHook
+, cups
+, libcupsfilters
+, ghostscript
+, pkg-config
+, poppler_utils
+, mupdf # mutool
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libppd";
+  version = "2.0b2";
+
+  src = fetchFromGitHub {
+    owner = "OpenPrinting";
+    repo = "libppd";
+    rev = "2c650af8c8826a68c0c0293ab5511701eaac019a";
+    hash = "sha256-4KRJfzK/xeWlXmhUOlvGHsMmKqF6R4e5BiIA+h2UTII=";
+  };
+
+  configureFlags = [
+    "--disable-acroread"
+    "--localstatedir=/var"
+    "--sysconfdir=/etc"
+  ];
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [
+    cups
+    ghostscript
+    poppler_utils
+    mupdf.bin
+    libcupsfilters
+    zlib
+  ];
+
+  # Fix missing includes on darwin
+  prePatch = lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc")
+    ''
```

```suggestion
  postPatch = lib.strings.optionalString (stdenv.hostPlatform.libc != "glibc") ''
```
Always change code in postPatch otherwise patches might not apply.

Darwin is usually detected with stdenv.isDarwin.

Have you considered using a patch to make this easier to update for a new release?

---

> Darwin is usually detected with stdenv.isDarwin.

I wasn't quite sure in this case because the issue is that upstream relies on GNU-isms in the standard library, so it could technically also occur with other systems not based glibc. But obviously, the only tested non-glibc system is Darwin. So perhaps this level of detail isn't necessary?

> Have you considered using a patch to make this easier to update for a new release?

Yeah, I guess that would be the better option, the substituteInPlace are a bit hacky :slightly_smiling_face:

---

> So perhaps this level of detail isn't necessary?

I think it's good: `pkgsMusl.libppd` seems to eval. (can try building if needed)